src/client/mod.rs

```diff
@@ -1419,9 +1419,9 @@ impl IotHubClient {
             _ => error!("c_d2c_confirmation_callback({trace_id}): received confirmation from iothub with unknown IOTHUB_CLIENT_CONFIRMATION_RESULT"),
         }
 
-        tx_confirm.send(succeeded).expect(&format!(
-            "c_d2c_confirmation_callback({trace_id}): cannot send confirmation result"
-        ));
+        tx_confirm.send(succeeded).unwrap_or_else(|_| {
+            panic!("c_d2c_confirmation_callback({trace_id}): cannot send confirmation result")
```

Comment:
can you remind we, why we want to panic in a library?
we take the possibility from the application to catch that.

Reply:
because there is no other option since we're in a azure-iot-sdk-c callback. we could just log an error and wait until the timeout occurs. but even in this case we can only log an error. the application cannot catch that case and we should not wait blocking imho